pkg/operator/vttablet/pod.go

```diff
@@ -71,6 +74,14 @@ func UpdatePod(obj *corev1.Pod, spec *Spec) {
 	// Update labels, but ignore existing ones we don't set.
 	update.Labels(&obj.Labels, spec.Labels)
 
+	// Update annotations and recreate pod if annotations are updated or removed.
```

Comment:
We were already setting annotations in `UpdatePodInPlace`. We need to stop doing it there and move it to here, since now we are relying on annotation changes to cause a rolling update of Pods.

Other than that, the comments from `etcd/pod.go` apply in this file too.

Reply:
Makes sense. Done.